deploy/sre-prometheus/100-managed-velero-operator.PrometheusRule.yaml

```diff
@@ -0,0 +1,50 @@
+apiVersion: monitoring.coreos.com/v1
+kind: PrometheusRule
+metadata:
+  labels:
+    prometheus: sre-managed-velero-operator-alerts
+    role: alert-rules
+  name: sre-managed-velero-operator-alerts
+  namespace: openshift-monitoring
+spec:
+  groups:
+  - name: sre-managed-velero-operator-alerts
+    rules:
+    - alert: VeleroPodOfflineSRE
+      expr: kube_deployment_status_replicas_available{namespace="openshift-velero",deployment="velero"} == 0
+      for: 15m
+      labels:
+        severity: warning
+        namespace: openshift-monitoring
+      annotations:
+        message: Velero pod has been missing or unavailable for 15m
+    - alert: VeleroHourlyObjectBackupMissed
+      expr: |
+        velero_backup_attempt_total{namespace="openshift-velero",schedule="hourly-object-backup"} > 0 and
+        time() - velero_backup_last_successful_timestamp{namespace="openshift-velero",schedule="hourly-object-backup"} > 3600
+      for: 30m
```

Comment:
It would be great to have some comments to outline what these are actually doing.

If I read this right.. we are verifying that we've at least attempted a backup, and that the "time since last successful" is greater than 1 hour.

Why do we have the 30 minute pending delay on that?

Reply:
The delay is there to add wiggle room for backup jobs, on the assumption that one backup may take a little longer than the previous to complete.  The delay is arbitrary but consistent: half the backup frequency (hourly -> 30m / daily -> 12h / weekly -> 84h).

So for example if an hourly backup hasn't completed in the last hour-and-a-half, assume something's wrong and raise an alert.

Those values can certainly be tweaked; it was just an initial guess.

Reply:
The duration of jobs should be fairly consistent. The way that `for` is used here is "the alert needs to be active for this long before we take action". We want to leave enough time for the backup to complete, but we can find that out with something like `rate(velero_backup_duration_seconds_sum{namespace="openshift-velero",schedule="hourly-object-backup"}[5h])/rate(velero_backup_duration_seconds_count{namespace="openshift-velero",schedule="hourly-object-backup"}[5h])` (right now it looks like about 60 seconds, but the daily/weekly might be a bit longer with the snapshot API calls).